### ISSUE TITLE
Inconsistent dice score between training and evaluate dice score

### ISSUE THREAD

**Issue by minhnhat93** [baumgach/acdc_segmenter#7]
Hello,

Nice works. I am trying to use this code for one of my projects. I tried running the code with default setting for the modified 2d unet with cross entropy loss. However, I got inconsistent dice score for during training and evaluating for the validation set like this:

Dice score for training and validation set during training
![image](https://user-images.githubusercontent.com/8745417/53142716-31c38b80-3552-11e9-86ab-30c2c535c255.png)

Dice score for validation set using evaluate_patients.py
![image](https://user-images.githubusercontent.com/8745417/53142849-a696c580-3552-11e9-9db1-5fb2a7baa287.png)

Is there any reason for this difference? Also, I looked into some sample output from the model and I get the feeling that the dice score for the training set should be closer to almost 100% rather than around 90% like in the Tensorflow summary. Thank you in advance!

**Comment by baumgach** [comment]
I think the training dice always tends to be a bit higher because slight overfitting may occur. Perhaps also it is due to a random difference in validation and training sets causing the training set to be slightly easier. I never looked into this in detail. 

The training and validation in tensorboard happens on 2D slices. However, the final dice is calculated by stacking the predictions into a 3D volume and then calculating the Dice score on the whole volume. 3D Dice scores are always larger than 2D because you are counting more voxels/pixels and a single pixel deviation is relatively less harmful to the total score.
